test/acceptance/test/utils.go

```diff
@@ -508,3 +514,41 @@ func mergePR(repoAbsolutePath, prLink string) {
 	Expect(err).ShouldNot(HaveOccurred())
 	Eventually(session).Should(gexec.Exit())
 }
+
+func createAppReplicas(repoAbsolutePath string, appManifestFilePath string, replicasSetValue int, workloadName string) string {
+	log.Infof("Editing app-manifest file in git repo to create replicas of workload: %s", workloadName)
+	appManifestFile := repoAbsolutePath + "/" + appManifestFilePath
+	_ = runCommandPassThrough([]string{}, "sh", "-c", fmt.Sprintf("sed -ie 's/replicas: 1/replicas: %d/g' %s", replicasSetValue, appManifestFile))
+	changedValue, _ := runCommandAndReturnStringOutput(fmt.Sprintf("cat %s", appManifestFile))
+	return changedValue
+}
+
+func waitForReplicaCreation(namespace string, replicasSetValue int, timeout time.Duration) error {
+
+	replica := strconv.Itoa(replicasSetValue)
+	pollInterval := 5
+	if timeout < 5*time.Second {
+		timeout = 5 * time.Second
+	}
+	timeoutInSeconds := int(timeout.Seconds())
+
+	for i := pollInterval; i < timeoutInSeconds; i += pollInterval {
```

Comment:
there is a function in utils doing this already called `WaitUntil`  it also does retries, the advantage is that it will return the error if it keeps failing in the period of time defined by the timeout. So then we can assert it and it will print out the error if it is different from what are expecting.